### PR TITLE
NEWRELIC-3627 Improve agent error logging with source maps …

### PR DESCRIPTION
**Pull request by dsellarsnr**
NEWRELIC-3627 Improve agent error logging with source maps
    - For agent use `import "source-map-support/register"` on entrypoints
    - For web, browser version of source-map-support did not work due to browser CORS restrictions when it tries to load the sourcemap from a file:// uri scheme. Instead, the agent handles mapping with the source map in `errorReporter.ts`
    - To make source mapping work same across IDEs, make name of webview consistent in all IDEs - JB was in the minority so webview.ts renamed to index.ts.

**This PR Addresses:**  
[NEWRELIC-3627 Improve agent error logging with source maps ](https://issues.newrelic.com/browse/NEWRELIC-3627)  

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

File: shared/agent/src/errorReporter.ts [hunk 1]
```diff
@@ -71,7 +169,15 @@ class NewRelicErrorReporterProvider
 		} else {
 			hash = request.message;
 		}
-		cacheKey = md5(hash || "");
+		return md5(hash ?? "");
```
Comment (bcanzanella) [inline shared/agent/src/errorReporter.ts:172]: didn't know we had a null coalescing operator

File: shared/agent/src/errorReporter.ts [hunk 2]
```diff
@@ -48,20 +54,112 @@ export class ErrorReporter {
...
+
 class NewRelicErrorReporterProvider
 	extends ErrorReporterProviderBase
 	implements IErrorReporterProvider
 {
+	private readonly stackMapper?: StackMapper.StackMapper;
+
 	constructor(private agent: CodeStreamAgent, session: CodeStreamSession) {
 		super(session);
+		const webSourceMap = path.resolve(__dirname, WEB_SOURCE_MAP);
```
Comment (bcanzanella) [inline shared/agent/src/errorReporter.ts:76]: any risk of anything in here throwing an error?

**Review by bcanzanella — APPROVED** [review]

File: shared/agent/src/errorReporter.ts (hunk 1 above)
Comment (dsellarsnr) [reply shared/agent/src/errorReporter.ts:172]: it is fairly new and less problematic - explicitly checks for null / undefined whereas || would be triggered for falsey "", 0 , false, etc.

File: shared/agent/src/errorReporter.ts (hunk 2 above)
Comment (dsellarsnr) [reply shared/agent/src/errorReporter.ts:76]: Yah, wrapping in try / cache

**Review by bcanzanella — APPROVED** [review]

**Pull request merged**